File: seunggon/week4/special2.py
def solution(arr1, arr2):
    m = len(arr1)      
    p = len(arr1[0])  
    n = len(arr2[0])   

    answer = [[0] * n for _ in range(m)]

    for i in range(m):       
        for j in range(n):    
            total = 0        
            for k in range(p): # arr1의 열과 arr2의 행
                total += arr1[i][k] * arr2[k][j]
            
            answer[i][j] = total

    return answer

File: dongwoo/week4/special2.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];
        int x = arr1.length;
        int y = arr1[0].length;
        int p = arr2.length;
        int k = arr2[0].length;

        for (int i = 0; i < x; i++) {
            for (int j = 0; j < k; j++) {
                int t = 0;
                for (int u = 0; u < y; u++) {
                    t += (arr1[i][u] * arr2[u][j]);
                }
                answer[i][j] = t;
            }
        }
        return answer;
    }
}
